hack/generator/pkg/astmodel/armconversion/convert_from_arm_function_builder.go

```diff
@@ -224,6 +225,99 @@ func (builder *convertFromARMBuilder) ownerPropertyHandler(
 	return []dst.Stmt{result}
 }
 
+func (builder *convertFromARMBuilder) flattenedPropertyHandler(
+	toProp *astmodel.PropertyDefinition,
+	fromType *astmodel.ObjectType) []dst.Stmt {
+
+	if !toProp.WasFlattened() {
+		return nil
+	}
+
+	for _, fromProp := range fromType.Properties() {
+		if toProp.WasFlattenedFrom(fromProp.PropertyName()) {
+			return builder.buildFlattenedAssignment(toProp, fromProp)
+		}
+	}
+
+	panic("couldn’t find source ARM property that k8s property was flattened from")
+}
+
+func (builder *convertFromARMBuilder) buildFlattenedAssignment(toProp *astmodel.PropertyDefinition, fromProp *astmodel.PropertyDefinition) []dst.Stmt {
+	if len(toProp.FlattenedFrom()) > 1 {
+		// this doesn't appear to happen anywhere in the JSON schemas currently
+		panic("need to implement multiple levels of flattening!")
+	}
+
+	allTypes := builder.codeGenerationContext.GetAllReachableTypes()
+
+	// the from shape here must be:
+	// 1. maybe a typename, pointing to…
+	// 2. maybe optional, wrapping …
+	// 3. maybe a typename, pointing to…
+	// 4. an object type
+
+	// (1.) resolve any outer typename
+	fromPropType, err := allTypes.FullyResolve(fromProp.PropertyType())
+	if err != nil {
+		panic(err)
+	}
+
+	var fromPropObjType *astmodel.ObjectType
+	var objOk bool
+	// (2.) resolve any optional type
+	if fromPropOptType, ok := fromPropType.(*astmodel.OptionalType); ok {
+		// TODO: need to generate a nil check in the generated code?
+		// (3.) resolve any inner typename
+		elementType, err := allTypes.FullyResolve(fromPropOptType.Element())
+		if err != nil {
+			panic(err)
+		}
+
+		// (4.) resolve the inner object type
+		fromPropObjType, objOk = elementType.(*astmodel.ObjectType)
+	} else {
+		// (4.) resolve the inner object type
+		fromPropObjType, objOk = fromPropType.(*astmodel.ObjectType)
+	}
+
+	if !objOk {
+		panic("property marked as flattened from non-object type, which doesn’t make sense")
+	}
+
+	// *** Now generate the code! ***
+	toPropName := toProp.PropertyName()
+	nestedProp, ok := fromPropObjType.Property(toPropName)
+	if !ok {
+		panic("couldn't find source of flattened property")
+	}
+
+	stmts := builder.typeConversionBuilder.BuildConversion(astmodel.ConversionParameters{
```

Comment:
minor: maybe put the `astmodel.ConversionParameters` onto a new line? It gets a bit lost at the end of this one IMO